op-batcher/batcher/driver.go

```diff
@@ -272,10 +294,21 @@ func (l *BatchSubmitter) loop() {
 	// start the receipt/result processing loop
 	receiptLoopDone := make(chan struct{})
 	defer close(receiptLoopDone) // shut down receipt loop
+
+	var txpoolState atomic.Int32
+	txpoolState.Store(TxpoolGood)
 	go func() {
 		for {
 			select {
 			case r := <-receiptsCh:
+				if errors.Is(r.Err, txpool.ErrAlreadyReserved) && txpoolState.CompareAndSwap(TxpoolGood, TxpoolBlocked) {
+					l.Log.Info("incompatible tx in txpool")
```

Comment:
Having the swap be part of the condition is technically fine, but makes it a little hard to follow -- short-circuiting from the `errors.Is` is how this check avoids setting the state to "Blocked", but could we put that *inside* the condition? Are there cases where the error is `ErrAlreadyReserved`, but the state is *not* "Good"?

Reply:
Not sure I follow the concern about readability.  This seemed a natural way to me to have it print out the warning exactly once per cancellation attempt. Perhaps related: yes, ErrAlreadyReserved can be returned in any of the three states.  Imagine having published a bunch of transactions all at once of type calldata when there is a blocking blob tx.  Each would likely result in ErrAlreadyReserved here, with only the first receipt flipping the state from Good to Blocked.  For the remaining txs, the state would most likely be Blocked (but could be pending or even good depending on the timing of the cancellation tx handling.)